backend/routes.py

```diff
@@ -4,26 +4,115 @@
 
 
 # Import all the necessary modules and packages
-from flask import current_app, request, jsonify, make_response
+# from flask import abort
+import jwt
+import uuid
+from flask import current_app, json, request, jsonify, make_response, abort
 from app import create_app, db
 from models import Login, User, Project
-
+from functools import wraps
+from werkzeug.security import generate_password_hash, check_password_hash
+# import werkzeug
 
 # Create an application instance
 app = create_app()
 
 
+def token_required(f):
+    @wraps(f)
+    def decorator(*args, **kwargs):
+
+        token = None
+
+        if 'x-access-tokens' in request.headers:
+            token = request.headers['x-access-tokens']
+
+        if not token:
+            data = {'message': 'Valid token is missing'}
+            response = make_response(jsonify(data=data, status=400))
+            return response
+
+        try:
+            data = jwt.decode(token, app.config['SECRET_KEY'], algorithms=["HS256"])
+            # return jsonify({'token': jwt.decode(token, app.config['SECRET_KEY'], algorithms=["HS256"])})
+            current_user = Login.query.filter_by(public_id=data['public_id']).first()
+            return f(current_user, *args, **kwargs)
+        except:
+            data = {'message': 'Invalid token'}
+            response = make_response(jsonify(data=data, status=400))
+            return response
+    return decorator
+
+
+# API to register new user
+@app.route("/register", methods=["POST"], strict_slashes=False)
+def register_user():
+    email = request.json.get("email") # Test if this is the right way to get info
+    password = request.json.get("password")
+    
+    if not email or not password:
+        abort(400, description="Empty email or password")
+    
+    if Login.query.filter_by(email = email).first() != None:
+        abort(400, description="Existing user")
+    
+    if not '@minerva.kgi.edu' in email and not 'minerva.edu' in email:
```

Comment:
There is a way to handle authentication much more easier if you use Google cloud.

If you host this project on gcloud, you can use the Identiy-Aware Proxy to limit users of Minerva only to be able to access the app
https://cloud.google.com/iap

It is quite simple in that you only need to toggle a switch and it is turned on.

Reply:
I see. Thanks for the suggestion. I'll look into it and implement it in future PR :D

Reply:
For sure, it simplifies a lot of the work and you don't need to write much code if at all.
High security and a custom app for your needs